lib/redux/middlewares.dart

```diff
@@ -19,68 +19,40 @@ void apiMiddleware(
 
   if (action is FetchOrganizationsAction) {
     try {
-      final response = await api.organizations();
-      if (response.statusCode == 200) {
-        final responseJson = json.decode(response.body) as List;
-        final orgList = List<Map<String, dynamic>>.from(responseJson);
-        final orgs = orgList.map((Map<String, dynamic> r) => Organization.fromJson(r)).toList();
-        store.dispatch(FetchOrganizationsSuccessAction(orgs));
-        store.dispatch(SelectOrganizationAction(orgs.first));
-        store.dispatch(
-            FetchProjectsAction(store.state.globalState.selectedOrganization));
-      } else {
-        store.dispatch(FetchOrganizationsFailureAction());
-      }
+      final organizations = await api.organizations();
+        store.dispatch(FetchOrganizationsSuccessAction(organizations));
+        store.dispatch(SelectOrganizationAction(organizations.first));
+        store.dispatch(FetchProjectsAction(organizations.first));
     } catch (e) {
-      store.dispatch(FetchOrganizationsFailureAction());
+      store.dispatch(FetchOrganizationsFailureAction(e as Error));
```

Comment:
wondering if we should check first if `e is Error`, in Kotlin we could do `as?` and it'd not fail if the cast fails, but I guess this is not possible on Dart

Reply:
Isn't it more likely is `e is Exception` to be true as opposed to `e is Error`?

`e` could be `String`, `Exception`, `Error`, not sure what else but attempting to cast to `Error` alone here might not be the way to go

Reply:
the custom exceptions written by this PR `extends Error`:

`class ApiError extends Error`

also, `Error` has a `stackTrace` field, `Exception` doesn't.

`FetchOrganizationsFailureAction` ctor has an `Error` param, so I believe `e is Error` is the correct one if the thrown error is our own, but if the exception is thrown by any other reason (no network available) or something, it could be `Exception` or even just a random object so maybe its worth creating a method extension to convert this properly, checking its type by `Error`, `Exception`, any object or even null

Reply:
btw I am not fully sure if custom exceptions should be Error or Exception, I found both examples on Google, but I believe it should have a trade-off which I am unaware of right now

Reply:
You are correct right, in the case that not one of our errors is thrown, we would loose the information. -.- I changed the actions to hold a dynamic property, as the handling of the type is not done here and right now we don't do anything. Will use the property later to react to certain errors in reducers or to log them in a middleware. And would do the the method extension then to handle all possible cases if thats ok with you.